test/system/src/test/java/io/pravega/test/system/TransactionStressTest.java

```diff
@@ -0,0 +1,388 @@
+            FutureHelpers.allOf(readerFutureList).get();
+
+            log.info("All writers have stopped. Setting Stop_Read_Flag. Event Written Count:{}, Event Read " +
+                    "Count: {}", eventWriteCount.get(), eventsReadFromPravega.size());
+            assertEquals(eventWriteCount.get(), eventsReadFromPravega.size());
+            assertEquals(eventWriteCount.get(), new TreeSet<>(eventsReadFromPravega).size()); //check unique events.
+
+            //close all the writers
+            log.info("Closing writers");
+            writerList.forEach(writer -> writer.close());
+            //close all readers
+            log.info("Closing readers");
+            readerList.forEach(reader -> reader.close());
+            //delete readergroup
+            log.info("Deleting readergroup {}", readerGroupName);
+            readerGroupManager.deleteReaderGroup(readerGroupName);
+        }
+        //seal the stream
+        CompletableFuture<Boolean> sealStreamStatus = controller.sealStream(scope, STREAM_NAME);
+        log.info("Sealing stream {}", STREAM_NAME);
+        assertTrue(sealStreamStatus.get());
+        //delete the stream
+        CompletableFuture<Boolean> deleteStreamStatus = controller.deleteStream(scope, STREAM_NAME);
+        log.info("Deleting stream {}", STREAM_NAME);
+        assertTrue(deleteStreamStatus.get());
+
+        //delete the scope
+        CompletableFuture<Boolean> deleteScopeStatus = controller.deleteScope(scope);
+        log.info("Deleting scope {}", scope);
+        assertTrue(deleteScopeStatus.get());
+        log.info("Test {} succeeds ", "MultiReaderWriterTxnWithFailOver");
+    }
+
+
+    private CompletableFuture<Void> startReading(final ConcurrentLinkedQueue<Long> readResult, final AtomicLong writeCount, final
+    AtomicLong readCount, final AtomicBoolean exitFlag, final EventStreamReader<Long> reader) {
+        return CompletableFuture.runAsync(() -> {
+            log.info("Exit flag status {}", exitFlag);
+            log.info("Read count {} and write count {}", readCount.get(), writeCount);
+            while (!(exitFlag.get() && readCount.get() == writeCount.get())) {
+                log.info("Entering read loop");
+                // exit only if exitFlag is true  and read Count equals write count.
+                try {
+                    final Long longEvent = reader.readNextEvent(SECONDS.toMillis(60)).getEvent();
+                    log.debug("Reading event {}", longEvent);
+                    if (longEvent != null) {
+                        //update if event read is not null.
+                        readResult.add(longEvent);
+                        readCount.incrementAndGet();
+                        log.debug("Event read count {}", readCount);
+                    } else {
+                        log.debug("Read timeout");
+                    }
+                } catch (Throwable e) {
+                    log.warn("Test Exception while reading from the stream: ", e);
```

Comment:
This needs to fail the test. This will require setting an exception on the test state mentioned above.